doc/source/user_guide.rst

```diff
@@ -0,0 +1,38 @@
+.. _user_guide:
+
+.. currentmodule:: nomenclature
+
+
+User Guide
+==========
+
+There are two main classes that the user interacts with when using the nomenclature
```

Comment:
Start here with the main use cases: validation against a codelist and region-processing (the use case, not the class description).

Reply:
Restructured it accordingly.